src/main/java/org/apache/mesos/scheduler/recovery/constrain/TimedLaunchConstrainer.java

```diff
@@ -10,29 +10,33 @@
  * rate-limiting purposes.
  */
 public class TimedLaunchConstrainer implements LaunchConstrainer {
-    private AtomicLong lastLaunch;
+    private AtomicLong lastPermanentRecoveryLaunch;
```

Comment:
It'd make sense to mark this with the unit, ie "lastPermanentRecoveryLaunchMillis" or "lastPermanentRecoveryLaunchMs", to avoid mixup with minDelaySeconds.

For that matter it'd probably make sense to store minDelaySeconds as milliseconds, so that the units for comparison are consistent throughout.